pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/GenericRow.java

```diff
@@ -49,7 +49,17 @@
  */
 public class GenericRow {
 
+  /**
+   * This key is used by a Decoder/RecordReader to handle 1 record to many records flattening.
+   * If a Decoder/RecordReader produces multiple GenericRows from the given record, they must be put into the destination GenericRow as a List<GenericRow> with this key
+   * The segment generation drivers handle this key as a special case and process the multiple records
+   */
   public static final String MULTIPLE_RECORDS_KEY = "$MULTIPLE_RECORDS_KEY$";
+  /**
+   * This key is used by the FilterTransformer to handle filtering out of records during ingestion
+   * The FilterTransformer puts this key into the GenericRow with value true, if the record matches the filtering out criteria, based on FilterConfig
+   */
+  public static final String FILTER_RECORD_KEY = "$FILTER_RECORD_KEY$";
```

Comment:
I would suggest renaming these as "$PINOT_INTERNAL_MULTIPLE_RECORDS_KEY$" and "$PINOT_INTERNAL_SHOULD_INGEST_ROW$". Of course, this will mean that the decoders that put in the existing MULTIPLE_RECORDS_KEY should be handled right (if we released this in 0.4.0 without documentation). I don't see any documentation for it in our docs. Let me know if I am not looking at the right place.

Reply:
Multiple records key feature didn't go into the release. It has been documented here: https://docs.pinot.apache.org/developers/advanced/ingestion-level-transformations#one-record-into-many

Renamed to `$SKIP_RECORDS_KEY$`. Also based on Jackie's comment, will avoid making it long, so multiple key remains `$MULTIPLE_RECORDS_KEY$`